vyper/codegen/expr.py

```diff
@@ -536,6 +536,14 @@ def parse_BinOp(self):
                     pos=pos,
```

Comment:
i think we should remove the above logic for handling literals and try to rely solely on the info passed down from the typechecker.